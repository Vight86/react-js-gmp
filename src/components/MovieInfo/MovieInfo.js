import Logo from '../Logo/Logo';
import Button from '../Button/Button';
import Title from '../Title/Title';
import s from './MovieInfo.scss';

const MovieInfo = ({ movieInfo, closeMovieInfo }) => {
  const {
    poster, title, rating, genre, year, time, description,
  } = movieInfo;

  return (
    <header className={s.movieInfoHeader}>
      <div className={s.headerTop}>
        <Logo />
        <Button
          theme="search"
          isWide={false}
          onClick={closeMovieInfo}
        />
      </div>
      <div className={s.movieInfo}>
        <figure className={s.poster}>
          <img
            src={poster}
            alt={title}
          />
        </figure>
        <Title className={s.title}>{title}</Title>
        <div className={s.rating}>{rating}</div>
        <div className={s.genre}>{genre}</div>
        <div className={s.year}>{year}</div>
        <div className={s.time}>{time}</div>
        <p className={s.description}>{description}</p>
      </div>
    </header>
  );
};

MovieInfo.propTypes = {
  movieInfo: PropTypes.shape({
    poster: PropTypes.string,
    title: PropTypes.string,
    rating: PropTypes.number,
    genre: PropTypes.string,
    year: PropTypes.number,
    time: PropTypes.string,
    description: PropTypes.string,
  }),
  closeMovieInfo: PropTypes.func.isRequired,
};

MovieInfo.defaultProps = {
  movieInfo: {},
};

export default MovieInfo;
