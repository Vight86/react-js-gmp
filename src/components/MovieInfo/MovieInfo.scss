@use '../../scss/variables';

.movieInfoHeader {
  max-width: variables.$containerWidth;
  padding: 0 2em;
  font-size: 1rem;
  margin: 0 auto;
  display: block;

}

.headerTop {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-top: 3.5em;
  padding-bottom: 3em;
  font-size: inherit;
}

.movieInfo {
  font-family: variables.$font-primary;
  font-size: 1rem;
  padding: 0 2em;
  display: grid;
  grid-template-columns: minmax(35%, 325px) auto auto;
  grid-template-rows: auto auto auto 1fr;
  grid-template-areas:
    "poster title rating"
    "poster genre genre"
    "poster year year"
    "poster description description";
  box-sizing: border-box;
  margin-bottom: 2em;

  @media screen and (max-width: variables.$bp-large) {
    font-size: 0.8rem;
  }

  @media screen and (max-width: variables.$bp-small) {
    width: 100%;
    grid-template-columns: 90% auto;
    grid-template-areas:
    "poster poster"
    "title rating"
    "genre genre"
    "year year"
    "description description";
  }
}

.poster {
  grid-area: poster;
  font-size: 1rem;
  margin-right: 5em;

  & > img {
    width: 100%;
    object-fit: cover;
  }

  @media screen and (max-width: variables.$bp-small) {
    margin-right: 0;
    margin-bottom: 2em;
  }
}

.title {
  grid-area: title;
}

.rating {
  margin-top: -0.4em;
  font-family: inherit;
  font-size: 2em;
  width: 3em;
  height: 3em;
  line-height: 3em;
  color: variables.$color-fivefold;
  grid-area: rating;
  border: 1px solid variables.$color-fivefold;
  border-radius: 50%;
  text-align: center;
}

.genre {
  grid-area: genre;
  margin-top: 1em;
  font-family: inherit;
  font-style: normal;
  font-weight: 500;
  font-size: 1.4em;
  color: variables.$color-fivefold;
  opacity: 0.5;
}

.year, .time {
  margin-top: 1.3em;
  grid-area: year;
  color: variables.$color-primary;
  font-family: inherit;
  font-weight: 300;
  font-size: 2.4em;
}

.time {
  margin-left: 4.2em;
}

.description {
  margin-top: 1.5em;
  grid-area: description;
  font-family: inherit;
  font-style: normal;
  font-weight: 300;
  font-size: 2em;
  line-height: 1.2;
  color: variables.$color-fivefold;
  opacity: 0.5;
}
