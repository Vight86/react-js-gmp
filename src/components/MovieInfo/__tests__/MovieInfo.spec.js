import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import MovieInfo from '../MovieInfo';

const mockCloseMovieInfo = jest.fn();
const mockMovie = {
  poster_path: 'localhost',
  title: 'test title',
  vote_average: 9,
  genres: ['crime', 'drama'],
  release_date: '1999-01-01',
  runtime: 120,
  overview: 'Test overview',
};

describe('MovieInfo', () => {
  const setup = () => render(<MovieInfo
    movieInfo={mockMovie}
    closeMovieInfo={mockCloseMovieInfo}
  />);

  test('should render movie info', () => {
    const { container } = setup();
    expect(container.innerHTML).toMatchSnapshot();
  });

  test('should invoke close Movie Info callback', () => {
    const { container } = setup();
    fireEvent.click(container.querySelector('button'));
    expect(mockCloseMovieInfo).toHaveBeenCalled();
  });
});
