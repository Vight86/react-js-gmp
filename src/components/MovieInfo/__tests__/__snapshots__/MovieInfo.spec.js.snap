// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo should render movie info 1`] = `"<header class=\\"movieInfoHeader\\"><div class=\\"headerTop\\"><a href=\\"/\\"><img src=\\"[object Object]\\" title=\\"netflix roulette\\" alt=\\"netflix roulette logo\\"></a><button type=\\"button\\" class=\\"search normal\\"></button></div><div class=\\"movieInfo\\"><figure class=\\"poster\\"><img src=\\"localhost\\" alt=\\"test title\\"></figure><h1 class=\\"primary\\">test title</h1><div class=\\"rating\\">9</div><div class=\\"genre\\">crime, drama</div><div class=\\"year\\">1999</div><div class=\\"time\\">2h 0min</div><p class=\\"description\\">Test overview</p></div></header>"`;
